[docs] landing page (fixes #21750)

maxpumperla opened the pull request:
Preview: [docs](https://ray--21859.org.readthedocs.build/en/21859/)

More informative and attractive landing page with a focus on

- [x] Three core personas with clearer messaging
- [x] A better overview of what Ray is and what you can do with it
- [x] More instructive "get involved"
- [x] Adding a "what docs are for you?" guide

![Screenshot 2022-01-25 at 16 40 49](https://user-images.githubusercontent.com/3462566/151009292-d634867a-2cc0-43c5-ac85-45154815d8a1.png)

maxpumperla commented:
@richardliaw @ericl @edoakes wording matters here, please give me all your nitpicks.

richardliaw commented:
A couple stylistic changes to request:

<img width="1057" alt="Screen Shot 2022-01-25 at 11 17 54 AM" src="https://user-images.githubusercontent.com/4529381/151044276-f89eee25-172f-490d-b823-86e180e45ee1.png">

Can we avoid overlapping into the right sidebar?
---------------

<img width="1134" alt="Screen Shot 2022-01-25 at 11 18 40 AM" src="https://user-images.githubusercontent.com/4529381/151044385-8db4f3be-90bc-4a92-a624-848e2811f608.png">
Can we make the buttons consistent?

richardliaw commented on doc/source/index.md:
```diff
@@ -0,0 +1,208 @@
+
+Ray is an open-source project developed at UC Berkeley RISE Lab.
+As a general-purpose and universal distributed compute framework, you can flexibly run any compute-intensive Python workload — from distributed training or hyperparameter tuning to deep reinforcement learning and production model serving.
+To help you, Ray Core provides a simple, universal API for building distributed applications.
+And Ray's native libraries and tools enable you to run complex ML applications with Ray.
+You can deploy these applications on any of the major cloud providers, including AWS, GCP, and Azure, or run them on your own servers.
+Ray also has a growing [ecosystem of community integrations](ray-overview/ray-libraries), including [Dask](https://docs.ray.io/en/latest/data/dask-on-ray.html), [MARS](https://docs.ray.io/en/latest/data/mars-on-ray.html), [Modin](https://github.com/modin-project/modin), [Horovod](https://horovod.readthedocs.io/en/stable/ray_include.html), [Hugging Face](https://huggingface.co/transformers/main_cl ...
+The following figure gives you an overview of the Ray ecosystem.
```
Right now this reads as a big paragraph; can we insert some linespaces?

richardliaw commented on doc/source/index.md:
```diff
@@ -0,0 +1,208 @@
+
+**Build distributed applications with <img src="ray-overview/images/ray_svg_logo.svg" alt="ray" width="50px">Core**
+^^^
+Ray Core provides a [simple and flexible API](ray-core/walkthrough.rst) for building and running your distributed applications.
+You can often parallelize single machine code with little to zero code changes.
+
+For instance, with Ray Core you can easily parallelize your functions using 
+[Ray Tasks](ray-core/walkthrough.rst#remote-functions-tasks) or your classes using 
+[Ray Actors](ray-core/actors.rst). 
```
```suggestion
Ray Core provides a [simple and flexible API](ray-core/walkthrough.rst) for building and running your distributed applications.
You can easily parallelize your functions using 
[Ray Tasks](ray-core/walkthrough.rst#remote-functions-tasks) or your classes using 
[Ray Actors](ray-core/actors.rst). 
```

richardliaw commented on doc/source/index.md:
```diff
@@ -0,0 +1,208 @@
+---
+**Run machine learning workflows with <img src="ray-overview/images/ray_svg_logo.svg" alt="ray" width="50px">ML**
+^^^
+With Ray ML you can run tasks like [distributed data loading and compute](data/dataset.rst),
```
```
Ray ML is a toolkit for distributed machine learning. It provides libraries for distributed data processing, model training, tuning, reinforcement learning, model serving, and more. 
```

Perhaps we can add links to each, but I think we should condense the wording

richardliaw commented on doc/source/index.md:
```diff
@@ -0,0 +1,208 @@
+---
+**Deploy large-scale workloads with <img src="ray-overview/images/ray_svg_logo.svg" alt="ray" width="50px">Cluster**
+^^^
+With Ray Cluster you can deploy your workloads on [AWS, GCP, Azure](cluster/quickstart) or 
```
```suggestion
With a Ray cluster, you can deploy your workloads on [AWS, GCP, Azure](cluster/quickstart) or 
```

richardliaw commented on doc/source/index.md:
````diff
@@ -0,0 +1,208 @@
+```
+
+---
+**Deploy large-scale workloads with <img src="ray-overview/images/ray_svg_logo.svg" alt="ray" width="50px">Cluster**
````
```suggestion
**Deploy large-scale workloads with <img src="ray-overview/images/ray_svg_logo.svg" alt="ray" width="50px">Clusters**
```

richardliaw replied:
Also consider removing the svg? Not sure if it scales well across devices.

maxpumperla replied:
It's the size of the font, which does not change. can remove it later if people hate it.

richardliaw commented on doc/source/index.md:
````diff
@@ -0,0 +1,208 @@
+
+```{link-button} cluster/quickstart
+:type: ref
+:text: Get Started with Ray Cluster
````
I'm not sure about this. Can you give feedback on this particular element @edoakes @ericl ?

maxpumperla commented:
@richardliaw I addressed your comments and added a getting started guide today. This should be good to go now.

![Screenshot 2022-01-26 at 15 53 36](https://user-images.githubusercontent.com/3462566/151186758-88bfd01d-e7df-41d7-806c-144de7d19c53.png)
 
![Screenshot 2022-01-26 at 15 53 46](https://user-images.githubusercontent.com/3462566/151186798-77e3cb64-3b9e-4693-b2e3-568908e4317d.png)

richardliaw commented on doc/source/_toc.yml:
```diff
@@ -99,32 +100,19 @@ parts:
         title: "Tutorials and Examples"
 
-  - caption: Deploying Ray Clusters
+  - caption: Ray Clusters | Deployments
```
FYI, the caption says "Clusters" here but elsewhere we have "Ray Cluster", should we make that consistent?

richardliaw reviewed (approved):
Looks good to me!
